sanic/protocols/http2.py

```diff
@@ -0,0 +1,134 @@
+import asyncio
+import io
+import collections
+from functools import partial
+from typing import List, Tuple
+
+from h2.connection import H2Connection
+from h2.events import (
+    ConnectionTerminated, DataReceived, RequestReceived, StreamEnded
+)
+from h2.errors import ErrorCodes
+from h2.exceptions import ProtocolError
+
+from sanic.request import Request
+from sanic.log import log
+
+RequestData = collections.namedtuple('RequestData', ['headers', 'data'])
+
+
+def generate_http2_protocol(ssl, parent_class):
+    ssl.set_ciphers("ECDHE+AESGCM")
+    ssl.set_alpn_protocols(["h2", "http/1.1"])
+
+    _HTTP2Protocol = type("HTTP2Protocol", (HTTP2Checker, parent_class), {})
+    _HTTP2Protocol.protocol_class = type("H2Protocol",
+                                         (Http2Protocol, parent_class), {})
+
+    return _HTTP2Protocol
+
+
+class HTTP2Checker:
+    def connection_made(self, transport):
+        ssl_object = transport.get_extra_info('ssl_object')
+        protocol = ssl_object.selected_alpn_protocol()
+
+        if protocol == 'h2':
+            self.__class__ = self.protocol_class
+            self.connection_made(transport)
+        else:
+            super().connection_made(transport)
+
+
+class Http2Protocol(asyncio.Protocol):
+
+    def connection_made(self, transport):
+        # TODO: Request timeouts
+        self.h2_connection = H2Connection(client_side=False)
```

Comment:
While the explicit kwarg here is supported now it's deprecated as of v2.5.0 in favour of an [explicit configuration object](https://python-hyper.org/h2/en/stable/api.html#configuration). This may provide a few other useful switches as well.